src/store/store.js

```diff
@@ -339,11 +339,21 @@ export default new Vuex.Store({
 
 			return true
 		},
-		async updateContext({ state, commit, dispatch }, { id, data }) {
+		async updateContext({ state, commit, dispatch }, { id, data, receivers }) {
 			let res = null
+			let share = {
+				nodeType: 'context',
+				nodeId: id,
+				displayMode: 2,
+			}
 
 			try {
 				res = await axios.put(generateOcsUrl('/apps/tables/api/2/contexts/' + id), data)
+				for (const receiver of receivers) {
+					share.receiverType = receiver.isUser ? 'user' : 'group'
+					share.receiver = receiver.user
+					await axios.post(generateUrl('/apps/tables/share'), share)
```

Comment:
@blizzz  When I try to share, I get the following error. Where am I going wrong?

<details>
<summary>Error </summary>
{"reqId":"X2W4HIOuNjA5wyUt7MLP","level":3,"time":"2024-04-11T08:00:56+00:00","remoteAddr":"192.168.21.5","user":"admin","app":"index","method":"POST","url":"/index.php/apps/tables/share","message":"Typed property OCA\\Tables\\Db\\ContextNavigation::$shareId must not be accessed before initialization in file '/var/www/html/lib/public/AppFramework/Db/Entity.php' line 108","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","version":"29.0.0.5","exception":{"Exception":"Exception","Message":"Typed property OCA\\Tables\\Db\\ContextNavigation::$shareId must not be accessed before initialization in file '/var/www/html/lib/public/AppFramework/Db/Entity.php' line 108","Code":0,"Trace":[{"file":"/var/www/html/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[["OCA\\Tables\\Controller\\ShareController"],"create"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Tables\\Controller\\ShareController","create",["OC\\AppFramework\\DependencyInjection\\DIContainer"],["tables.share.create"]]},{"file":"/var/www/html/lib/base.php","line":1069,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/tables/share"]},{"file":"/var/www/html/index.php","line":39,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","Line":170,"Previous":{"Exception":"Error","Message":"Typed property OCA\\Tables\\Db\\ContextNavigation::$shareId must not be accessed before initialization","Code":0,"Trace":[{"file":"/var/www/html/lib/public/AppFramework/Db/Entity.php","line":167,"function":"setter","class":"OCP\\AppFramework\\Db\\Entity","type":"->","args":["shareId",[6]]},{"file":"/var/www/html/apps/tables/lib/Service/ShareService.php","line":240,"function":"__call","class":"OCP\\AppFramework\\Db\\Entity","type":"->","args":["setShareId",[6]]},{"file":"/var/www/html/apps/tables/lib/Controller/ShareController.php","line":77,"function":"create","class":"OCA\\Tables\\Service\\ShareService","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/apps/tables/lib/Controller/Errors.php","line":16,"function":"OCA\\Tables\\Controller\\{closure}","class":"OCA\\Tables\\Controller\\ShareController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/apps/tables/lib/Controller/ShareController.php","line":78,"function":"handleError","class":"OCA\\Tables\\Controller\\ShareController","type":"->","args":[["Closure"]]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":232,"function":"create","class":"OCA\\Tables\\Controller\\ShareController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":138,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[["OCA\\Tables\\Controller\\ShareController"],"create"]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[["OCA\\Tables\\Controller\\ShareController"],"create"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Tables\\Controller\\ShareController","create",["OC\\AppFramework\\DependencyInjection\\DIContainer"],["tables.share.create"]]},{"file":"/var/www/html/lib/base.php","line":1069,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/tables/share"]},{"file":"/var/www/html/index.php","line":39,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/public/AppFramework/Db/Entity.php","Line":108},"message":"Typed property OCA\\Tables\\Db\\ContextNavigation::$shareId must not be accessed before initialization in file '/var/www/html/lib/public/AppFramework/Db/Entity.php' line 108","exception":{},"CustomMessage":"Typed property OCA\\Tables\\Db\\ContextNavigation::$shareId must not be accessed before initialization in file '/var/www/html/lib/public/AppFramework/Db/Entity.php' line 108"}}
</details>

Reply:
My bad :pensive: fix incoming

Reply:
thank you!